### PR TITLE
Add NYC taxi data tutorial

### LINKED ISSUE #44: Add poster tutorial with NYC taxi data to docs

### LINKED ISSUE #53: Better names in test data creation functions
- rename `dense_data_array` to `data_array`
- rename `with_variance` agument to something else

### PR DESCRIPTION
**Pull request by nvaytet**
Add example of NYC taxi data visualization from Euroscipy poster.

Fixes #44 

Also fix #53 at the same time

### CODE REVIEW

File: docs/custom-interfaces/gallery/nyc_taxi.ipynb [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    "in_node = pp.input_node(data)\n",
+    "in_node.name = 'Input data'\n",
+    "\n",
+    "# Slider node that provides index to slice\n",
+    "slider = ipw.IntSlider(min=0, max=23)\n",
+    "slider_node = pp.widget_node(slider)\n",
+    "slider_node.name = 'Slider'\n",
+    "\n",
+    "# Node that actually does the slicing\n",
+    "slice_node = pp.node(lambda da, ind: da['hour', ind])(da=in_node, ind=slider_node)\n",
```
Comment (jl-wynen) [inline docs/custom-interfaces/gallery/nyc_taxi.ipynb:69]: This line barely fits when my browser is full screen.

File: docs/custom-interfaces/gallery/nyc_taxi.ipynb [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+   "outputs": [],
+   "source": [
+    "# Input data node\n",
+    "in_node = pp.input_node(data)\n",
+    "in_node.name = 'Input data'\n",
+    "\n",
+    "# Slider node that provides index to slice\n",
+    "slider = ipw.IntSlider(min=0, max=23)\n",
+    "slider_node = pp.widget_node(slider)\n",
+    "slider_node.name = 'Slider'\n",
```
Comment (jl-wynen) [inline docs/custom-interfaces/gallery/nyc_taxi.ipynb:66]: Can you add a label that indicates the the slider is 'hour'?

File: docs/index.rst [hunk 3]
```diff
@@ -69,11 +69,12 @@ Get in touch
...
 
 .. toctree::
-   :caption: Tutorials
+   :caption: Building custom interfaces
    :maxdepth: 3
    :hidden:
 
-   tutorials/nodes-tutorial
+   custom-interfaces/introduction
+   custom-interfaces/gallery
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/index.rst:77]: I think yes to all. But then you should expand on the code comments a little.

**Comment by nvaytet** [comment]
Annoyingly, I cannot get the title of the gallery notebook to show up...
Screenshot at 2022-10-27 10-52-01

compared to the nbsphinx doc page
Screenshot at 2022-10-27 10-52-51

File: docs/examples/gallery.ipynb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "6a6ec63f-51df-4bca-ab29-762f6d13420c",
+   "metadata": {
+    "tags": [
+     "nbsphinx-gallery"
+    ]
+   },
+   "source": [
+    "* [NYC Taxi visualization](gallery/nyc-taxi.ipynb)"
```
Comment (jl-wynen) [inline docs/examples/gallery.ipynb:20]: Please rename to something that describes the purpose of this example.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**